pkg/controller/update_cluster.go

```diff
@@ -584,21 +599,25 @@ func (c *M3DBController) findPodInstanceToRemove(
 		return nil, nil, pkgerrors.WithMessage(err, "cannot sort pods")
 	}
 
-	for i := len(podIDs) - 1; i >= 0; i-- {
+	var (
+		podsToRemove      []*corev1.Pod
+		instancesToRemove []placement.Instance
+	)
+	for i := len(podIDs) - 1; i >= desiredInstanceCount; i-- {
```

Comment:
nit: Maybe it's worth moving the check that `desiredInstanceCount` is greater than zero into this function just in case we call it from other places in the future?